src/pass/storage_rewrite.cc

```diff
@@ -718,10 +718,10 @@ class StoragePlanRewriter : public IRMutator {
                     src_entry->attach_scope_ == thread_scope_ &&
                     src_entry->elem_type == ae.alloc->type.element_of() &&
                     visitor.Check(s.stmt, var, src)) {
-                  uint64_t const_nbits = static_cast<uint64_t>(
```

Comment:
This change is non-necessary, and makes the code worse. It is best to do the multiplication in uint32_t and then cast the result to uint64_t and then do the math in uint64_t, which is more expensive. Hypothetically when multiply 32 bits integers the result could overflow 32 bits range. But first it is not going to happen. Second if it happens you have a bigger problem as you cannot allocate 4GB (2^32) worth of memory at a time. So don't change here.

Reply:
Wait, it's `nbits`, not bytes, so it's "only" 536 MB (that's probably why it's `uint64_t` and not `size_t` in the first place). Still, the multipliers are probably small enough, as your first point mentions.